### ISSUE TITLE
Compilation issues with new master version

### ISSUE THREAD

**Issue by AlbertoEAF** [lemire/fast_double_parser#42]
Hello @lemire :)

For the work in https://github.com/microsoft/LightGBM/pull/3405 you suggested updating to master which I attempted but it fails with multiple definition of some functions. I don't know if they're now defined in the header or there are missing header guards but from a green build using the old commit e09e5aad828bcb16bea7ed0ed8322e019112fdbe on my pc now I get these errors at linking time:

```
[ 98%] Linking CXX shared library ../lib_lightgbm.so
CMakeFiles/_lightgbm.dir/src/boosting/boosting.cpp.o: In function `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)':
boosting.cpp:(.text+0x1040): multiple definition of `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)'
CMakeFiles/_lightgbm.dir/src/application/application.cpp.o:application.cpp:(.text+0x1e0): first defined here
CMakeFiles/_lightgbm.dir/src/boosting/gbdt.cpp.o: In function `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)':
gbdt.cpp:(.text+0x4a00): multiple definition of `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)'
CMakeFiles/_lightgbm.dir/src/application/application.cpp.o:application.cpp:(.text+0x1e0): first defined here
CMakeFiles/_lightgbm.dir/src/boosting/gbdt_model_text.cpp.o: In function `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)':
gbdt_model_text.cpp:(.text+0x3d50): multiple definition of `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)'
CMakeFiles/_lightgbm.dir/src/application/application.cpp.o:application.cpp:(.text+0x1e0): first defined here
CMakeFiles/_lightgbm.dir/src/boosting/gbdt_prediction.cpp.o: In function `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)':
gbdt_prediction.cpp:(.text+0x0): multiple definition of `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)'
CMakeFiles/_lightgbm.dir/src/application/application.cpp.o:application.cpp:(.text+0x1e0): first defined here
CMakeFiles/_lightgbm.dir/src/io/bin.cpp.o: In function `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)':
bin.cpp:(.text+0x4af0): multiple definition of `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)'
CMakeFiles/_lightgbm.dir/src/application/application.cpp.o:application.cpp:(.text+0x1e0): first defined here
CMakeFiles/_lightgbm.dir/src/io/config.cpp.o: In function `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)':
config.cpp:(.text+0x270): multiple definition of `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)'
CMakeFiles/_lightgbm.dir/src/application/application.cpp.o:application.cpp:(.text+0x1e0): first defined here
CMakeFiles/_lightgbm.dir/src/io/config_auto.cpp.o: In function `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)':
config_auto.cpp:(.text+0x1a0): multiple definition of `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)'
CMakeFiles/_lightgbm.dir/src/application/application.cpp.o:application.cpp:(.text+0x1e0): first defined here
CMakeFiles/_lightgbm.dir/src/io/dataset.cpp.o: In function `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)':
dataset.cpp:(.text+0x1840): multiple definition of `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)'
CMakeFiles/_lightgbm.dir/src/application/application.cpp.o:application.cpp:(.text+0x1e0): first defined here
CMakeFiles/_lightgbm.dir/src/io/dataset_loader.cpp.o: In function `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)':
dataset_loader.cpp:(.text+0x1530): multiple definition of `fast_double_parser::Emulate64x64to128(unsigned long&, unsigned long, unsigned long)'
(...)
```

Do I need to change anything in my CMakeLists / includes to use the new version?

This is the new MR https://github.com/feedzai/LightGBM/pull/8 where the latest stable version for fast_double_parser will be pulled into the MR https://github.com/microsoft/LightGBM/pull/3405.

**Comment by lemire** [comment]
Yes. It should be easy to fix. A backup function is not declared static when it should. Let me first build a test case so we can reproduce.

**Comment by lemire** [comment]
Fixed in version 0.2.0. Please use tag v0.2.0.

 https://github.com/lemire/fast_double_parser/releases/tag/v0.2.0
